notifications/views.py

```diff
@@ -30,11 +40,53 @@ def mark_all_as_read(self, request):
         count = notifications.update(is_read=True)
         return Response({"status": f"{count} notifications marked as read"})
 
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user.id)
+
 
 class NotificationPreferenceViewSet(viewsets.ModelViewSet):
     queryset = NotificationPreference.objects.all()
     serializer_class = NotificationPreferenceSerializer
-    permission_classes = [IsAuthenticated]
+    permission_classes = [IsAuthenticated, IsOwner]
 
     def get_queryset(self):
         return self.queryset.filter(user=self.request.user)
+
+    def create(self, request, *args, **kwargs):
+
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        type_name = serializer.validated_data.get("type")
+        type_instance = Type.objects.filter(name=type_name).first()
+
+        if not type_instance:
+            raise ValidationError(f"Type '{type_name}' not found.")
+
+        duplicate_exists = NotificationPreference.objects.filter(
+            user=request.user, type=type_instance
+        ).exists()
+
+        if duplicate_exists:
+            raise ValidationError(
+                "A preference with this user and type already exists.[view]"
+            )
+
+        serializer.save(user=request.user, type=type_instance)
+
+        headers = self.get_success_headers(serializer.data)
+        return Response(
+            serializer.data, status=status.HTTP_201_CREATED, headers=headers
+        )
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
```

Comment:
`perform_create()` is already saving the user inside `serializer.save(user=self.request.user)`.
`create()` manually calls `serializer.save(user=request.user, type=type_instance)`, which is redundant.
Move the `type_instance` logic to `perform_create()`, and let `create()` remain clean.